docs/build-notifications.rst

```diff
@@ -20,31 +20,256 @@ Take these steps to enable build notifications using email:
+as long as it is valid JSON. Below you have a couple of examples,
+and in the following section you will find all the available variables.
+
+.. figure:: /_static/images/webhooks-payload.png
+   :width: 80%
+   :align: center
+   :alt: Custom payload
 
-The project name, slug and its details for the build will be sent as POST request to your webhook URL:
+   Custom payload
+
+Slack
++++++
+
+.. code-block:: json
+
+   {
+     "attachments": [
+       {
+         "color": "#db3238",
+         "blocks": [
+           {
+             "type": "section",
+             "text": {
+               "type": "mrkdwn",
+               "text": "*Read the Docs build failed*"
+             }
+           },
+           {
+             "type": "section",
+             "fields": [
+               {
+                 "type": "mrkdwn",
+                 "text": "*Project*: <{{project.url}}|{{project.name}}>"
+               },
+               {
+                 "type": "mrkdwn",
+                 "text": "*Version*: {{version.name}} ({{build.commit}})"
+               },
+               {
+                 "type": "mrkdwn",
+                 "text": "*Build*: <{{build.url}}|{{build.id}}>"
+               }
+             ]
+           }
+         ]
+       }
+     ]
+   }
+
+More information on `the Slack Incoming Webhooks documentation <https://api.slack.com/messaging/webhooks>`_.
+
+Discord
++++++++
 
 .. code-block:: json
 
+   {
+     "username": "Read the Docs",
+     "content": "Read the Docs build failed",
+     "embeds": [
        {
-            "name": "Read the Docs",
-            "slug": "rtd",
-            "build": {
-                "id": 6321373,
-                "commit": "e8dd17a3f1627dd206d721e4be08ae6766fda40",
-                "state": "finished",
-                "success": false,
-                "date": "2017-02-15 20:35:54"
-            }
+         "title": "Build logs",
+         "url": "{{build.url}}",
+         "color": 15258703,
+         "fields": [
+           {
+             "name": "*Project*",
+             "value": "{{project.url}}",
+             "inline": true
+           },
+           {
+             "name": "*Version*",
+             "value": "{{version.name}} ({{build.commit}})",
+             "inline": true
+           },
+           {
+             "name": "*Build*",
+             "value": "{{build.url}}"
+           }
+         ]
+       }
+     ]
+   }
+
+More information on `the Discord webhooks documentation <https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks>`_.
+
+Variable substitutions reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``{{event}}``
+  Event that triggered the webhook, one of ``build:triggered``, ``build:failed``, or ``build:passed``.
+
+``{{build.id}}``
+  Build ID.
+
+``{{build.commit}}``
+  Commit corresponding to the build, if present (otherwise ``""``).
+
+``{{build.url}}``
+  URL of the build.
+
+``{{build.docsurl}}``
```

Comment:
I feel like this isn't what we normally call this. We should standardize this language. I think it would probably be `docs_url`, but we should check if we're already exposing this, and document it in the glossary. 

Looks like we have `docs_url` in our API response: https://github.com/readthedocs/readthedocs.org/blob/d6f7347f82d7b3e210cd7e7e84d860282dd029f8/readthedocs/api/v2/serializers.py#L143

Reply:
/cc @stsewd

Reply:
I changed this to be `docs_url` in the other PR

https://github.com/readthedocs/readthedocs.org/blob/a2b6a2aa71bbf899534cbc9e58c45ea4d988b4af/readthedocs/projects/models.py#L1583-L1597

Reply:
This is also missing documenting `{{ build.start_date }}`

Reply:
Done!